coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py

```diff
@@ -267,42 +286,36 @@ def pathsOptimization(keys):
                 'body': '{"error":"operation '+operation+' is not implemented for AWS-CoioteDM integration"}'
             }
 
-        try:
-            credentials = getCoioteSecrets()
-        except Exception as ex:
-            print('Error: ' + str(ex))
-            return {
-                'statusCode': 400,
-                'body': '{"error":"' + str(ex) + '"}'
-            }
-        #uri=restUri+'/api/coiotedm/v3/tasksFromTemplates/deviceBlocking/'+thingName
+        # uri=restUri+'/api/coiotedm/v3/tasksFromTemplates/deviceBlocking/'+thingName
         """
         in this approach, the task is scheduled at Coiote - and then we are performing 2nd call to trigger session even if a device is deregistered
         as the task has exec condition that it is executed only when the device is registered
         below resolved by 2nd exec condition - if a device is in queue mode, the task is executed once the device comes with some LwM2M message:
         we can also add an optional parameter in the shadow to wait for a device (by default set to false) and in this case not performing 2nd call triggering session
         If instead these 2 the method commented above is used, Coiote will respond with error indicating that the device is deregistered and the task will not be scheduled at all
         """
-        uri = credentials.url+'/api/coiotedm/v3/tasksFromTemplates/device/'+thingName
-        headers = {'Content-Type':'application/json'}
-        apiCallResp = requests.post(uri, data=body, headers=headers, auth=(credentials.username, credentials.password))
-        qjResponseCode = apiCallResp.status_code
-        qjResponseBody = apiCallResp.text
-        if qjResponseCode != 201:
-            print('Error: Coiote DM responded with: ' + str(qjResponseCode) + ': ' + qjResponseBody)
+        for certificate, privateKey in getCertificateFiles():
+            uri = REST_URI+'/tasksFromTemplates/device/'+thingName
+            headers = {'Content-Type': 'application/json', 'Authorization': 'Certificate'}
+            apiCallResp = requests.post(uri, data=body, headers=headers, cert=(certificate, privateKey), verify=False, timeout=10)
```

Comment:
Nie żebym Cię podpuszczał, ale nie zmienisz `body` na słownik, tak żeby móc napisać
```suggestion
            headers = {'Authorization': 'Certificate'}
            apiCallResp = requests.post(uri, json=body, headers=headers, cert=(certificate, privateKey), verify=False, timeout=10)
```
i nie trzeba pisać headers 2 razy